frame/support/src/traits/metadata.rs

```diff
@@ -29,6 +29,10 @@ pub trait PalletInfo {
 	fn index<P: 'static>() -> Option<usize>;
 	/// Convert the given pallet `P` into its name as configured in the runtime.
 	fn name<P: 'static>() -> Option<&'static str>;
+	/// Convert the given pallet `P` into its containing crate name.
+	fn crate_name<P: 'static>() -> Option<&'static str>;
```

Comment:
Maybe we should update the doc of PalletInfo.

PalletInfo was introduced as: 
```
/// Provides information about the pallet setup in the runtime.
///
/// An implementor should be able to provide information about each pallet that
/// is configured in `construct_runtime!`.
```

It was meant to associate information given to construct_runtime for each pallets. The index and the name of the pallet are defined by construct_runtime, that is why there was a need to introduce this trait and the associated type in frame_system::Config::PalletInfo.

And PalletInfoAccess, was implemented to easily access the information in frame_system::Config::PalletInfo.

With this PR there is a bit of interdependency between both traits: PalletInfoAccess will call PalletInfo for name and index, and PalletInfo will call PalletInfoAccess for crate_name and crate_version.

I think it is fine, but the doc of PalletInfo and PalletInfoAccess should be updated to something like:
```
/// Provides information about the pallet itself and its setup in the runtime.
///
/// An implementor should be able to provide information about each pallet that
/// is configured in `construct_runtime!`.
pub trait PalletInfo {
```
```
/// Provides information about the pallet itself and its setup in the runtime.
///
/// Declare some information and access the information provided by [`PalletInfo`] for a specific
/// pallet.
pub trait PalletInfoAccess {
```

I'm ok to have this difficult relation as long as both traits are meant to be implemented by pallet macro and construct_runtime

Reply:
Good point. I think version should be moved to `PalletInfoAccess` and `PalletInfo` gives the name/module_path as I said above.